### ISSUE TITLE
Heading-level outline should not list headings inside <template> elements

### ISSUE THREAD

**Issue by simevidas** [validator/validator#309]
Example:

```html
  <h1>H1 heading</h1>
  
  <template>
    <h2>H2 heading inside template</h2>
  </template>
```

Demo: https://validator.w3.org/nu/?showoutline=yes&doc=https%3A%2F%2Foutput.jsbin.com%2Fputawin%2Fquiet

![capture](https://cloud.githubusercontent.com/assets/716405/16236204/cd18fcd0-37d7-11e6-909a-bda90c8a2692.PNG)

**Comment by sideshowbarker** [comment]
Fixed and pushed to https://checker.html5.org/ and https://validator.w3.org/nu

Thanks for catching this.
